boot/bootutil/src/loader.c

```diff
@@ -966,6 +966,10 @@ boot_copy_region(struct boot_loader_state *state,
                 if (off + bytes_copied < hdr->ih_hdr_size) {
                     /* do not decrypt header */
                     blk_off = 0;
+                    /* check for overflow, return BOOT_EBADIMAGE */
```

Comment:
`/* Check for overflow. */` is sufficient. Stating what the return does is redundant and just increases the likelihood that the code gets changed in the future and the comment missed.